backend/src/controllers/regionController.js

```diff
@@ -60,208 +150,124 @@
-        },
-    }));
-    let result_divisions = [];
-    for (let region of regions) {
-        const query = `
-            SELECT r.* FROM hierarchy_region_mapping hrm
-            JOIN regions r ON hrm.region_id = r.id
-            WHERE alt_region_id = :regionId AND hierarchy_id = :hierarchyId
-            `
-        const result = await sequelize.query(query, {
-            replacements: { regionId: region.regionId, hierarchyId: hierarchyId },
-            type: QueryTypes.SELECT,
-            mapToModel: true,
-            model: Region
-        });
-        result_divisions = result_divisions.concat(result.map(region => region.dataValues));
+    if (!region) {
+      return res.status(404).json({ message: 'Region not found' });
     }
-    return result_divisions;
-}
-
-// Retrieve subregions for a specific region
-getSubregions = async (regionId, hierarchyId, getAll) => {
-    try {
-        // Check if the region exists
-        const region = await Hierarchy.findOne({
-            where: {
-                regionId: regionId,
-                hierarchyId: hierarchyId
-            },
-        });
-
-        if (!region) {
-            return { data: [], message: 'Region not found', status: 404 };
-        }
 
-        // Retrieve subregions
-        let subregions;
-        // Check the getAll query parameter
-        if (getAll === 'true') {
-            subregions = await getAllSubregions(regionId, hierarchyId);
-        } else {
-            subregions = await Hierarchy.findAll({
-                where: { parentId: regionId, hierarchyId: hierarchyId },
-                mapToModel: true,
-                model: Hierarchy
-            });
-        }
-
-        if (subregions.length === 0) {
-            return { data: [], message: 'Region has no subregions', status: 202 };
-        }
-
-        return { data: subregions, status: 200 };
-    } catch (err) {
-        console.error(err);
-        return { data: [], message: 'Internal Server Error', status: 500 };
-    }
+    return res.status(200).json(region.toApiFormat());
+  } catch (err) {
+    console.error(err);
+    return res.status(500).json({ message: 'Internal Server Error' });
+  }
 };
 
 exports.getSubregions = async (req, res) => {
-    const regionId = req.params.regionId;
-    const getAll = req.query.getAll || false;
-    const hierarchyId = req.query.hierarchyId || 1;
+  const { regionId } = req.params;
+  const getAll = req.query.getAll || false;
+  const hierarchyId = req.query.hierarchyId || 1;
 
-    const subregions = await getSubregions(regionId, hierarchyId, getAll);
-    return res.status(subregions.status).json(subregions.data ? subregions.data.map(r => r.toApiFormat()) : { message: subregions.message });
-}
+  const subregions = await getSubregions(regionId, hierarchyId, getAll);
+  const result = subregions.data ? subregions.data.map(
+    (r) => r.toApiFormat(),
+  ) : { message: subregions.message };
+  return res.status(subregions.status).json(result);
```

Comment:
**Suggestion:** Use a more descriptive variable name than 'r' for better readability.
```suggestion
    const result = subregions.data ? subregions.data.map(
      (region) => region.toApiFormat(),
    ) : { message: subregions.message };
```